src/arrayMethodSort.js

```diff
@@ -4,8 +4,26 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const callback =
+      compareFunction === undefined
+        ? (value1, value2) => (`${value1}` > `${value2}` ? 1 : -1)
+        : compareFunction;
+
+    for (let j = 0; j < this.length - 1; j++) {
+      for (let i = 0; i < this.length - 1; i++) {
+        const returnCallback = callback(this[i], this[i + 1]);
+
+        if (returnCallback > 0) {
+          const saveValue = this[i + 1];
+
+          this[i + 1] = this[i];
+          this[i] = saveValue;
+        }
+      }
+    }
```

Comment:
The current implementation uses a bubble sort algorithm, which is correct but not efficient for large arrays. Consider using a more efficient sorting algorithm like quicksort or mergesort for better performance.